scrabble.js

```diff
@@ -1,8 +1,144 @@
-var Scrabble = function() {};
+Scrabble =  {
+  scoreChart: {
+    1: ["A","E", "I", "O", "U", "L", "N", "R", "S", "T"],
+    2: ["D", "G"],
+    3: ["B", "C", "M", "P"],
+    4: ["F", "H", "V", "W", "Y"],
+    5: ["K"],
+    8: ["J", "X"],
+    10: ["Q", "Z"]
+  },
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+
+  score: function(word) {
+    var score = 0;
+    var upWord = word.toUpperCase();
+    var num = Object.keys(Scrabble.scoreChart);
+    var letters = Object.values(Scrabble.scoreChart);
+    for (i = 0; i < upWord.length; i++) {
+      for (d = 0; d < num.length; d ++){
+        if (letters[d].includes(upWord[i])){
+          score += parseInt(num[d]);
+        }
+      }
+    }
+    if (word.length >= 7){
+      score += 50;
+    }
+    return score;
+  },
+
+  highestScore: function(words){
+    var scores = [];
+    var max = 0;
+    var maxWord = "";
+    words.forEach(function(element, index){
+      scores.push(Scrabble.score(element));
+    });
+    for (var i = 0; i < scores.length; i++){
+      if (scores[i] === max){
+        if( (words[i].length === 7 && words[i] > maxWord) || ( words[i].length < maxWord.length && maxWord.length !== 7)){
+          maxWord = words[i];
+        }
+      }
+      else  if (scores[i] > max) {
+        maxWord = words[i];
+        max = scores[i];
+      }
+    }
+    return maxWord;
+  }
+};
+
+var Player = function(name){
+  this.person = name;
+  this.plays = [];
+  this.scores = [];
+  this.finalScore = 0;
+};
+
+function add(a, b) {
+  return a + b;
+}
+
+Player.prototype.play = function(words){
+  this.plays.push(words);
+  if (this.hasWon()){
+    return false;
+  }
+  else {
+    this.scores.push(Scrabble.score(words));
+    this.totalScore();
+    return Scrabble.score(words);
```

Comment:
By calling `score` in here twice, you are doing the scoring logic twice which is unnecessary